ledger/participant-integration-api/src/main/scala/platform/index/ReadOnlySqlLedger.scala

```diff
@@ -112,22 +120,29 @@ private[platform] object ReadOnlySqlLedger {
         Some(enricher),
       )
 
+    def contractsStoreOwner(
+        lfValueTranslationCache: LfValueTranslation.Cache,
+        contractsReader: LedgerDaoContractsReader,
+    ): Resource[ContractStore] =
+      TranslationCacheBackedContractsStore
```

Comment:
is there a flag to turn this feature on/off?
if not I would advise to add one: until we have legacy schema and supporting code, we must be able not to use this, if we only want to build support for this in the append-only schema

Reply:
This is the legacy caching implementation. Nothing changed, it was just extracted out of the ContractsReader